### PR TITLE
pr/ccimx6ul qca6564: Add wireless support for ConnectCore 6UL SOM and SBCs

### PR DESCRIPTION
**Pull request by alexgg**
The ConnectCore 6UL SOM contains a Qualcomm QCA6564A chipset with the following features:

- Dual Band 5GHz 802.11ac or 2.4/5GHz 802.11n Wireless Local Area Network (US regulatory domain)
- Open, Wi-Fi Protected Access (WPA) and Wi-Fi Protected Access 2 (WPA2 (personal and enterprise) authentications
- Data encryption using Advanced Encryption Standard (AES) or Temporal Key Integrity Protocol (TKIP)
- Station and SoftAP concurrent modes
- Bluetooth (BT) 4.2

This commit adds software support for it.

### CODE REVIEW

File: recipes-kernel/kernel-modules/kernel-module-qca6564_git.bb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+SRCBRANCH = "qca6564/master"
+SRCREV = "2e780b29209ca6b0f71442be5fd36cf0ddddc661"
+
+SRC_URI = "\
+    git://github.com/digi-embedded/qcacld-2.0.git;protocol=git;branch=${SRCBRANCH} \
+    file://modprobe-qca6564.conf \
+"
+
+LOCALVERSION ?= "-${SRCBRANCH}"
```
Comment (otavio) [inline recipes-kernel/kernel-modules/kernel-module-qca6564_git.bb]: Please use:

PV = "...+${SRCPV}" 

so we keep track of the revision. The localversion seems useless here as this is a module.

File: recipes-kernel/kernel-modules/kernel-module-qca6564_git.bb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+EXTRA_OEMAKE += "BUILD_DEBUG_VERSION=0"
+# Flags for SDIO interface
+EXTRA_OEMAKE += "CONFIG_CLD_HL_SDIO_CORE=y"
+
+do_compile_prepend() {
+    export BUILD_VER=${PV}
+}
+
+do_install_prepend() {
+    sed -i -e "s/gVhtTxMCS=2/gVhtTxMCS=0/g" ${WORKDIR}/git/firmware_bin/WCNSS_qcom_cfg.ini
```
Comment (otavio) [inline recipes-kernel/kernel-modules/kernel-module-qca6564_git.bb]: Do this change on D, so we allow it to be run every time.

File: recipes-kernel/linux-firmware-qca6564/linux-firmware-qca6564_2.4-r2.2.bb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+SRC_URI = " \
+    file://nvm_tlv_3.2.bin \
+    file://rampatch_tlv_3.2.tlv \
+    file://bdwlan30_US.bin \
+    file://LICENCE.atheros_firmware \
+    file://otp30.bin \
+    file://qwlan30.bin \
+    file://utf30.bin \
+"
```
Comment (otavio) [inline recipes-kernel/linux-firmware-qca6564/linux-firmware-qca6564_2.4-r2.2.bb]: I am fine in you adding the recipe however we won't commit the binaries on the layer. Check if NXP does not has a tarball or Git for source, or put it on your github so we avoid putting binaries here.

File: recipes-kernel/kernel-modules/kernel-module-qca6564_git.bb (hunk 1 above)
Comment (alexgg) [reply recipes-kernel/kernel-modules/kernel-module-qca6564_git.bb]: I'll add ${SRCPV} to the PV variable above, thanks.

File: recipes-kernel/linux-firmware-qca6564/linux-firmware-qca6564_2.4-r2.2.bb (hunk 3 above)
Comment (alexgg) [reply recipes-kernel/linux-firmware-qca6564/linux-firmware-qca6564_2.4-r2.2.bb]: I will add them to a compressed tarball and push the changes.

File: recipes-kernel/kernel-modules/kernel-module-qca6564_git.bb (hunk 2 above)
Comment (alexgg) [reply recipes-kernel/kernel-modules/kernel-module-qca6564_git.bb]: Done

**Comment by alexgg** [comment]
Hi @otavio, I have addressed the comments, please let me know if there is anything missing from my side. Thanks!

**Comment by otavio** [comment]
@alexgg thank you; I will review it next week.

**Pull request merged**